chainer/functions/array/spatial_transformer_grid.py

```diff
@@ -34,7 +34,7 @@ def forward_cpu(self, inputs):
         return self._forward(inputs)
 
     def forward_gpu(self, inputs):
-        if not (cuda.cudnn_enabled and self.use_cudnn and
+        if not (cuda.cudnn_enabled and chainer.should_use_cudnn('>=auto') and
```

Comment:
I think `cuda.cudnn_enabled` is not necessary because `chainer.should_use_cudnn` is always False when cuDNN is not installed.
https://github.com/chainer/chainer/blob/master/chainer/__init__.py#L97

Also, the version of cuDNN can be checked by `lowest_version` arg for `should_use_cudnn`.